lib/graphiti/util/persistence.rb

```diff
@@ -69,43 +68,21 @@ def run
     persisted
   end
 
-  private
-
-  def add_hook(prc, lifecycle_event)
-    ::Graphiti::Util::TransactionHooksRecorder.add(prc, lifecycle_event)
-  end
-
-  # The child's attributes should be modified to nil-out the
-  # foreign_key when the parent is being destroyed or disassociated
-  #
-  # This is not the case for HABTM, whose "foreign key" is a join table
-  def update_foreign_key(parent_object, attrs, x)
-    return if x[:sideload].type == :many_to_many
+  def iterate(only: [], except: [])
```

Comment:
I was thinking If I should move this in Abstract adapter class

Because now this method is only being used from there, not from this class. But not sure if it's a good move considering the iterator relies on the instance variables of this class and we may end up passing the instance variables `@resource` and `@relationship` as arguments in some of the method call made from here to Abstract adapter class (e.g. in `process_belongs_to` and `process_has_many `).

Reply:
I think that's fine.